src/index.adoc

```diff
@@ -4,14 +4,542 @@ include::header.adoc[]
+to being simple (I can easily reason about what the code does).
+
+=== link:https://www.youtube.com/watch?v=f84n5oFoZBc[Rich Hickey: Hammock Driven
+Development^] Another great talk by Rich. The key message is simple:
+think before you code. Yet the talk is very influential and funny. The majority
+of people don't live this elementary rule, and they waste a lot of their time.
+
+=== link:https://www.youtube.com/watch?v=o9pEzgHorH0[Jack Diederich: Stop
+Writing Classes^] Despite its name, the talk doesn't imply that OOP is bad.
+Rather than that, it promotes writing short, readable code, avoiding premature
+generalizations, working with pure values, and avoiding complex OOP structures.
+It's funny and refreshing talk.
+
+=== link:https://www.youtube.com/watch?v=QM1iUe6IofM[Brian Will: Why OOP is Bad^]
+Brian gives a compelling argument that state encapsulation doesn't work: The
+real systems are dominated by cross-cutting concerns. When faced with the
+reality of real word programs, promises of encapsulation tend to be broken, and
+object graphs quickly become tangled webs.
+
+Apart from this, Brian does a very systematic breakdown of what Java has brought to engineering
+world, what are some good parts of it and why it thrives even though OOP is
+bad.
+
+=== link:https://www.youtube.com/watch?v=JjDsP5n2kSM&t=2169s[Jonathan Blow: How
+to program independent games^] You don't succeed by using fancy data structure or
+writing an algorithm that's a bit faster but buggy and fragile. Independent game
+developer needs to be "extremely effective at getting things done".
+
+=== Rich Hickey on immutable values versus references
+
+- link:https://www.infoq.com/presentations/Value-Values[The Value of Values^]
+- link:https://www.infoq.com/presentations/Are-We-There-Yet-Rich-Hickey[Are we
+  there yet^]
+
+This is Rich (again!) promoting immutable values and immutable data structures
+over references that change over time in a mutable manner. Also, Rich talks
+about the Epochal time model, where the whole (!) state of the program is
+represented by a single mutable reference that takes a sequence of immutable
+values, modeling a time-evolution of a state. This model is used with Redux
+framework.
+
+=== link:https://www.youtube.com/watch?v=FihU5JxmnBg[Stuart Halloway: Debugging
+with the Scientific Method^]
+
+Be less chaotic and more organized when debugging. State explicitly what's your
+hypothesis, test it, think. Write things down. Test one thing at a time. A good
+process is more important than tools; scientific approach with console.log is
+more powerful than ad-hoc trying random stuff with a strong debugger.
 
 == Tools
-include::tools/index.adoc[]
+
+=== Operating System
+Use either Mac or Linux for development. Don't use Windows, don't even think
+about it! There is always something working differently on Windows than on the
+other platforms. This costs us huge amounts of time debugging problems instead
+of developing valuable features.
```

Comment:
Well. I'm not sure it counts, quite a few people I know work from WSL = windows subsystem for linux. You basically get a virtualized linux inside of Windows, but the layer is very lightweight and you don't loose too much performance. Big bonus is that you avoid most of the driver issues.

Maybe would change it to "native" Windows, or mention that WSL2 is actually fine. I'm not saying I would recommend it over mac or linux, but it's much better than wasting days on broken laptop drivers and bad unstable UX that's often the case on modern linux desktop environments.